### DISCUSSION TITLE
How to load a config.yaml file for training?

### DISCUSSION THREAD

**Discussion by SilverStarCoder** [open-edge-platform/anomalib#1758] [Q&A]
In the older version of anomalib, I could load a `config.yaml `file using the `get_configurable_parameters` function from `anomalib.config`, but this is no longer available in the latest version. Will this be added in the future? Or alternatively, how else can I train using the config file, because this code doesn't work: `engine.fit(datamodule=datamodule, model=model, config="config.yaml")`. In the engine.py file, for fit, it says. If you have a ready configuration file, run it like this: `anomalib fit --config <config_file_path`, but surely that doesn't work as `config` is not a function argument.

**Comment by samet-akcay** [comment]
Hi @SilverStarCoder, yes, this feature is currently not available, but will be introduced. It will be something like;

```python
data.from_config()
model.from_config()
engine.from_config()
```
so each component could be instantiated from a config file. 

@ashwinvaidya17 and @harimkang, looks like this is of interest

**Reply by SilverStarCoder** [reply]
Thanks @samet-akcay for the response, so just double checking: I cannot at the moment train using a `config.yaml` file via API? I will have to input each one of the parameters (excluding the datamodule relevant ones) into the Engine class like this:
```
engine = Engine(
    callbacks=callbacks,
    pixel_metrics="AUROC",
    accelerator="auto", 
    devices=1,
    logger=False,
    ..........
)
```

**Reply by samet-akcay** [reply]
If you want to use the API, yes, this is the current approach. @ashwinvaidya17, do you want to add anything here?
